src/main/java/bio/terra/service/filedata/azure/tables/TableDirectoryDao.java

```diff
@@ -0,0 +1,247 @@
+
+    // Walk up the lookup directory path, finding missing directories we get to an
+    // existing one
+    // We will create the ROOT_DIR_NAME directory here if it does not exist.
+    String lookupDirPath = fileMetadataUtils.makeLookupPath(createEntry.getPath());
+    for (String testPath = lookupDirPath;
+        !testPath.isEmpty();
+        testPath = fileMetadataUtils.getDirectoryPath(testPath)) {
+
+      // !!! In this case we are using a lookup path
+      if (lookupByFilePath(tableServiceClient, datasetId, testPath) != null) {
+        break;
+      }
+
+      FireStoreDirectoryEntry dirToCreate = fileMetadataUtils.makeDirectoryEntry(testPath);
+      String partitionKey = getPartitionKey(datasetId, testPath);
+      String rowKey = encodePathAsAzureRowKey(testPath);
+      TableEntity entity = FireStoreDirectoryEntry.toTableEntity(partitionKey, rowKey, dirToCreate);
+      logger.info("Creating directory entry for {} in table {}", testPath, TABLE_NAME);
+      tableClient.createEntity(entity);
+    }
+
+    String fullPath = fileMetadataUtils.getFullPath(createEntry.getPath(), createEntry.getName());
+    String lookupPath = fileMetadataUtils.makeLookupPath(fullPath);
+    String partitionKey = getPartitionKey(datasetId, lookupPath);
+    String rowKey = encodePathAsAzureRowKey(lookupPath);
+    TableEntity createEntryEntity =
+        FireStoreDirectoryEntry.toTableEntity(partitionKey, rowKey, createEntry);
+    logger.info("Creating directory entry for {} in table {}", fullPath, TABLE_NAME);
+    tableClient.createEntity(createEntryEntity);
+  }
+
+  // true - directory entry existed and was deleted; false - directory entry did not exist
+  public boolean deleteDirectoryEntry(TableServiceClient tableServiceClient, String fileId) {
+    TableClient tableClient = tableServiceClient.getTableClient(TABLE_NAME);
+
+    // Look up the directory entry by id. If it doesn't exist, we're done
+    TableEntity leafEntity = lookupByFileId(tableServiceClient, fileId);
+    if (leafEntity == null) {
+      return false;
+    }
+
+    List<TableTransactionAction> deleteList = new ArrayList<>();
+    TableTransactionAction t =
+        new TableTransactionAction(TableTransactionActionType.DELETE, leafEntity);
+    deleteList.add(t);
+
+    FireStoreDirectoryEntry leafEntry = FireStoreDirectoryEntry.fromTableEntity(leafEntity);
+    String datasetId = leafEntry.getDatasetId();
+    String lookupPath = fileMetadataUtils.makeLookupPath(leafEntry.getPath());
+    while (!lookupPath.isEmpty()) {
+      // Count the number of entries with this path as their directory path
+      // A value of 1 means that the directory will be empty after its child is
+      // deleted, so we should delete it also.
+      ListEntitiesOptions options =
+          new ListEntitiesOptions()
+              .setFilter(
+                  String.format(
+                      "path eq '%s'", fileMetadataUtils.makePathFromLookupPath(lookupPath)));
```

Comment:
Should we do some sanitation here?  Could we end up with a case where the lookupPath was something like `' or 1 eq 1'` which would cause all of the things to delete?

Reply:
Hmm....I think we're pretty safe.  What you can put in the query is pretty limited I think

Reply:
We could do what I've done here (https://github.com/DataBiosphere/jade-data-repo/pull/1013/files#diff-4a47656762443349b2580bcfaddd2b56e5baaa32105229dd150af8ffe890668cR24) and bas64 encode it.

Reply:
I could see that going into a separate PR that does this for both firestore and storage tables